examples/dataflow-hashpipeline/scripts/hasher.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
```

Comment:
Having these utilities as CLI scripts seems to imply you expect the developer / user to manually verify the results. 
It would be much better to have an automated Integration Test that could be run on DirectRunner or TestDataflowRunner.

Think "I need to test the internal logic of how my pipeline interacts with various services and procduces a results but not how it reads / writes to pubsub":
- stage some example files in a public GCS bucket
- Use TestPipeline
- use your beam.Create to mock some input (PCollection of gcs paths)
- (insert the guts of your pipeline)
- use PAssert that you counted the expected number of SSNs per file.  (instead of writing to pubsub)

Reply:
The validate method is just something I added for debug purposes. I don't disagree that we should have some sort of automated testing, but is that something that can be added as a subsequent PR?

Reply:
I think we should have some amount of test coverage in this PR.
Reasoning: this is an OSS repo and tests will establish a base line to protect against regression if you or other contributors want to improve implementation or add features.